sdl2promela/translator.py

```diff
@@ -1729,6 +1710,212 @@ def __generate_statement(
+        promelamodel.MemberAccess,
+    ],
+    element_type: type,
+    parts: List[sdlmodel.Expression],
+    part_types: List[type],
+    iterator_name: str,
+):
+    statements = []
+
+    length_field = sdlmodel.MemberAccess(
+        target, append_operator_helpers.create_sdl_length_member()
+    )
+
+    translated_length_field = append_operator_helpers.create_promela_length_access(
+        translated_target
+    )
+
+    for part, part_type in zip(parts, part_types):
+        if isinstance(part, sdlmodel.SequenceOf):
+            part_length = len(part.elements)
+            for index in range(part_length):
+                field = sdlmodel.ArrayAccess(target, length_field)
+                statements.extend(
+                    __generate_assignment(
+                        context,
+                        field,
+                        part.elements[index],
+                        element_type,
+                    )
+                )
+                statements.append(
+                    append_operator_helpers.create_increment_step(
+                        translated_length_field
+                    )
+                )
+        else:
+            for_loop_builder = ForLoopBuilder()
+            for_loop_builder.withIterator(
+                VariableReferenceBuilder(iterator_name).build()
+            )
+            for_loop_builder.withFirst(0)
+            if part_type.Min == part_type.Max:
+                # this is a sequence of with fixed size
+                for_loop_builder.withLast(int(part_type.Max))
+            else:
+                for_loop_builder.withLast(
+                    append_operator_helpers.create_promela_length_access(
+                        __generate_expression(context, part)
+                    )
+                )
+
+            left_element = sdlmodel.ArrayAccess(target, length_field)
+            left_element.type = element_type
+            right_element = sdlmodel.ArrayAccess(
+                part, sdlmodel.VariableReference(iterator_name)
+            )
+            right_element.type = element_type
+            body = []
+            body.extend(
+                __generate_assignment(
+                    context,
+                    left_element,
+                    right_element,
+                    element_type,
+                )
+            )
+            body.append(
+                append_operator_helpers.create_increment_step(translated_length_field)
+            )
+            for_loop_builder.withBody(body)
+
+            statements.append(for_loop_builder.build())
+
+    return statements
+
+
+def __generate_assignment_with_append(
+    context: Context,
+    left: Union[
+        sdlmodel.VariableReference, sdlmodel.ArrayAccess, sdlmodel.MemberAccess
+    ],
+    right: sdlmodel.Expression,
+    left_type: type,
+):
+    destination = __generate_variable_name(context, left, True)
+    destination_type = resolve_asn1_type(context.sdl_model.types, left.type)
```

Comment:
This method is too long. Please split into descriptively named submethods.

Reply:
I've done some refactor. One function was extracted and I created a helper function, so creation of assignment statement is shorter and more readable IMO.